.github/workflows/build.yml

```diff
@@ -168,44 +219,95 @@ jobs:
           printf "\n\nSystem environment:\n\n"
           env
 
+      - name: Load Go Module Cache
+        uses: actions/cache@v4
+        with:
+          path: |
+            ${{ env.GOMODCACHE }}
+          key: ${{ env.CACHE_PREFIX }}-modcache-${{ hashFiles('**/go.mod') }}-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ env.CACHE_PREFIX }}-modcache
+
+      # Both load & update the cache when on default branch
+      - name: Load Go Build & Test Cache
+        id: go-cache
+        uses: actions/cache@v4
+        if: github.ref_name == github.event.repository.default_branch && github.event_name != 'pull_request'
+        with:
+          path: |
+            ${{ env.GOCACHE }}
+          key: ${{ env.CACHE_PREFIX }}-cache-${{ hashFiles('**/go.mod') }}-${{ hashFiles('**/go.sum') }}-${{ github.run_id }}
+          restore-keys: |
+            ${{ env.CACHE_PREFIX }}-cache
+
+      # Only load the cache when not on default branch
+      - name: Load Go Build & Test Cache
+        id: go-cache-restore
+        uses: actions/cache/restore@v4
+        if: github.ref_name != github.event.repository.default_branch || github.event_name == 'pull_request'
+        with:
+          path: |
+            ${{ env.GOCACHE }}
+          key: ${{ env.CACHE_PREFIX }}-cache-${{ hashFiles('**/go.mod') }}-${{ hashFiles('**/go.sum') }}-${{ github.run_id }}
+          restore-keys: |
+            ${{ env.CACHE_PREFIX }}-cache
+
       - name: Build rclone
         shell: bash
         run: |
           make
 
-      - name: Rclone version
+      - name: Print rclone Version
         shell: bash
         run: |
           rclone version
 
-      - name: Run tests
+      - name: Update mtime
+        shell: bash
+        run: |
+          go run ./bin/go-test-cache/main.go
```

Comment:
This is necessary because Golang caches the tests based on the modified timestamps of the file.

Reference: https://github.com/golang/go/issues/58571

It would be cleaner to do it with the newly added GOCACHEPROG but there's no compatible implementation for Github Actions Cache with that yet.